test/syscalls/linux/pty.cc

```diff
@@ -1148,6 +1148,23 @@ TEST_F(PtyTest, SwitchNoncanonToCanonNoNewlineBig) {
   ExpectFinished(replica_);
 }
 
+// If the canonical input buffer is empty, and we switch to non-canonical
+// mode, the input buffer should not be readable.
+TEST_F(PtyTest, SwitchCanonToNoncanonEmptyInput) {
+  constexpr char kInput[] = "";
+  ASSERT_THAT(WriteFd(master_.get(), kInput, sizeof(kInput)),
+              SyscallSucceedsWithValue(sizeof(kInput)));
```

Comment:
Sounds good. My other PR used `strlen`, which is why I was unsure: https://github.com/google/gvisor/pull/9956/files#r1475131791

Reply:
> Since we don't initialize kInput, strlen() may run off searching for null byte.

Sorry, this is wrong, it is initialized to "" and so there is a null byte. and sizeof() will return 1 while strlen() will return 0.

IIUC, in your other PR, you intend to write "nothing". So strlen there makes sense. Here you intend to write "empty buffer" (only /0 byte) so sizeof would make sense.